Introduce a basic check tool for verify lmcache env and config work as expected

maobaolong opened the pull request:
# Test result
- test
```bash
export LMCACHE_CONFIG_FILE=/Users/msy/projects/LMCache/examples/kv_cache_reuse/remote_backends/fs/example.yaml 
python lmcache/v1/basic_check.py --mode test
```
```
Performance Results:
----------------------------------------------------------------------------------------------------
| Operation            |     Avg (ms) |     Max (ms) |     Min (ms) |   Pass/All |  Pass Rate |
----------------------------------------------------------------------------------------------------
| EXISTS (non-exist)   |     0.358992 |     0.549500 |     0.251708 |        5/5 |     100.0% |
| PUT                  |     1.192058 |     2.032667 |     0.820458 |        5/5 |     100.0% |
| EXISTS (exist)       |     0.174992 |     0.201333 |     0.151042 |        5/5 |     100.0% |
| GET                  |     0.358725 |     0.428041 |     0.300042 |        5/5 |     100.0% |
----------------------------------------------------------------------------------------------------
```

- gen data
```bash
export LMCACHE_CONFIG_FILE=/Users/msy/projects/LMCache/examples/kv_cache_reuse/remote_backends/fs/example.yaml 
python lmcache/v1/basic_check.py --mode gen 
```


```
(base) ➜  ~ ll /tmp/lmcache_fs_test     
total 522240
-rw-r--r--@ 1 msy  wheel   1.8M  9 24 16:21 vllm@-dopkvstore_test-@3@123@-197b427d20f6eb42.data
-rw-r--r--@ 1 msy  wheel    64K  9 24 15:20 vllm@-dopkvstore_test-@3@123@-22eea9581ea71b2c.data
-rw-r--r--@ 1 msy  wheel    64K  9 24 15:14 vllm@-dopkvstore_test-@3@123@-26ae7377a9da95da.data
```

maobaolong committed:
Introduce a basic check tool for verify lmcache env and config work as expected

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

maobaolong committed:
Make it easy to be extended

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

maobaolong commented:
@sammshen @chunxiaozheng Would you like to take a look at this PR? Thanks!

maobaolong committed:
Refactor

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

maobaolong commented:
@sammshen 

Looks like fetch vllm failed in CI.

```

#28 1.009 error: Failed to fetch: `https://wheels.vllm.ai/nightly/vllm-0.11.0rc2.dev142%2Bg6e30010d2-cp38-abi3-manylinux1_x86_64.whl`
--
  | #28 1.009   Caused by: HTTP status client error (404 Not Found) for url (https://wheels.vllm.ai/nightly/vllm-0.11.0rc2.dev142%2Bg6e30010d2-cp38-abi3-manylinux1_x86_64.whl)
  | ------


```

chunxiaozheng reviewed (approved):
Nice job! LGTM!

sammshen commented on lmcache/v1/check/check_mode_test_storage_manager.py:
```diff
@@ -0,0 +1,105 @@
+
+async def async_get_storage_manager(storage_manager, key):
+    """Async wrapper for storage manager get method"""
+    return storage_manager.get(key)
```
why is there no await if this is async? what's the point of this being async

maobaolong replied:
Added

sammshen commented on lmcache/v1/check/check_mode_test_storage_manager.py:
```diff
@@ -0,0 +1,105 @@
+)
+
+
+async def async_contains_storage_manager(storage_manager, key):
```
why is there no await if this is async? what's the point of this being async

maobaolong replied:
added

sammshen commented on lmcache/v1/basic_check.py:
```diff
@@ -0,0 +1,74 @@
+    parser.add_argument(
+        "--mode",
+        required=True,
+        help="Operation mode (e.g. test, gen). Use 'list' to show available modes",
```
what are test and gen? sorry I don't quite understand?

maobaolong replied:
There are 3 sub command supported for now.
- test_remote: test the remote backend level basic performance, and print result.
- test_storage_manager: test the storage manager level basic performance, and print result.
- gen: generate kvcache data to the storage backend for testing the performance under a specified storage usage.

sammshen commented:
can we add something to the `examples/` folder with more explanation? :)

sammshen commented:
and/or some docs :)

maobaolong committed:
Add asyncio and use await for async method

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

maobaolong committed:
Fix style

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

maobaolong committed:
Add document and example

Signed-off-by: baoloongmao <baoloongmao@tencent.com>

maobaolong commented:
@sammshen Thanks for the review last week, sorry for the late update since I was in holiday.

Listing the main changes since last review:
- Updated the code to add await for async methods. And async is necessary since we can support `timeout` control by `asyncio.wait_for()` and reduce redundant code by keep the consistent code between `remote backend` and `storageManager`
- Add document
- Add example folder with a readme.md and a config file.

sammshen reviewed (commented):
Thanks @maobaolong ! LGTM!

maobaolong commented:
@sammshen @chunxiaozheng Thanks for your review!
